projects/js-packages/publicize-components/src/components/form/index.js

```diff
@@ -193,7 +221,10 @@ export default function PublicizeForm( {
 			shouldUploadAttachedMedia,
 		}
 	);
-	const shouldAutoConvert = isAutoConversionEnabled && isConvertible;
+	const shouldAutoConvert =
+		( isAutoConversionEnabled === undefined
+			? isAutoConversionEnabledInit
+			: isAutoConversionEnabled ) && isConvertible;
```

Comment:
Since we are kind of using initial state, then the store itself too, we have to have this logic, to use the initial state first, and the store after.